arxiv/metadata/metacheck.py

```diff
@@ -0,0 +1,687 @@
+    #
+    if contains_html_elements(v):
+        disposition = combine_dispositions(disposition, WARN)
+        complaints.append("Authors: contains HTML")
+    #
+    if not all_brackets_balanced(v):
+        disposition = combine_dispositions(disposition, WARN)
+        complaints.append("Authors: unbalanced brackets")
+    #
+    # Do I need to parse the authors list first?
+    if re.search( r";", v) and not re.search( "(.*;.*)", v):
+        disposition = combine_dispositions(disposition, WARN)
+        complaints.append("Authors: use commas, not ';' to separate authors")
+    #
+    # Authors list can NOT end with punctuation
+    if ends_with_punctuation(v):
+        disposition = combine_dispositions(disposition, WARN)
+        complaints.append("Authors: ends with punctuation")
+    #
+    
+    # Parse authors, check authors names - but not affiliations - for:
+    # ALL CAPS
+    # single word in all CAPS (not really done)
+    # [...] (in author name)
+    # Numbers (but Jennifer 8 Lee!)
+    # IEEE (confusable with affiliation)
+    # phd, prof, dr
+    # Single initial after parsing authors?
+    # TODO: letters after name (BUT Indian names like RS, etc?)
+    # TODO: extra curlies in TeX accents, e.g.: {\'e}
+    # (TODO: allow all Unicode ?)
+    # TODO: urls
+
+    # TODO: maybe remove duplicates from the complaints list
+    
+    parsed_authors = parse_author_affil(v) # => List[List[str]]
+    for author in parsed_authors:
+        # first is "keyname" second is fist name(s), third is suffix
+        keyname = author[0]
+        firstname = author[1]
+        suffix = author[2]
+        # Some quick special cases to skip:
+        if keyname.lower() == "author" and firstname == "" and suffix == "":
+            continue
+        if keyname.lower() == "authors" and firstname == "" and suffix == "":
+            continue
+        if keyname == ":" and firstname == "" and suffix == "":
+            continue
+        if firstname == "":
+            disposition = combine_dispositions(disposition, WARN)
+            complaints.append("Authors: lone surname")
+        #
+        # Don't reject Sylvie ROUX nor S ROUX
+        if re.match("^[A-Z]{3,}$", keyname) and \
+           re.match("^[A-Z]{3,}$", firstname):
+            disposition = combine_dispositions(disposition, WARN)
+            complaints.append("Author name is in all caps")
+        #
+        if re.match(r"^[A-Z]$", keyname) or \
+           re.match(r"^[A-Z]\.$", keyname):
+            disposition = combine_dispositions(disposition, WARN)
+            complaints.append("Author: found initial not surname")
+        #
+        # Reject e. e. cummings and "evans". Don't reject J von
+        if keyname == keyname.lower() and \
+           (firstname is None or firstname == firstname.lower()):
+            disposition = combine_dispositions(disposition, WARN)
+            complaints.append("Authors: no caps in name")

```

Comment:
Maybe change complaint to "Authors: no caps in keyname"?

Reply:
Do our users know what a "keyname" is?